### PR TITLE
Make adding/removing databases easier

### PR DESCRIPTION
**Pull request by nbrady-techempower**
This PR removes all database specific code from the toolset in the attempt to making adding and removing databases easier. This should allow you to simply add or remove a database folder without editing any toolset files.

You can think of each database folder now as a "plugin". The steps to add a new database to the TFB should look something like this now, using `redis` as an example

  * Create the folder `toolset/databases/redis`
  * Create `redis.dockerfile` in that folder. The dockerfile should install the database, build the worlds and fortunes table, and run the database.
  * Create a `redis.py` file. In it should be a `Database` class that implements `get_current_world_table()` and `test_connection()` static methods.
  * If python database drivers are needed for your new database, add them to the root toolset dockerfile here: https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/Dockerfile#L14

I also updated `metadata.py` so `tfb --new` will pull this dynamically and know about new databases.

TODO:
  - [x] - Wiki page on how to add a database (https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Development-Adding-a-Database)

### CODE REVIEW

File: toolset/databases/mongodb/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import pymongo
+import traceback
+
+from colorama import Fore
+from toolset.utils.output_helper import log
+
+class Database:
+    description = "A popular document-store database"
```
[the remark this answers is not in the document]
Comment (nbrady-techempower) [reply toolset/databases/mongodb/mongodb.py]: Description is used in metadata for tfb —new

[the remark this answers is not in the document]
Comment (msmith-techempower) [reply toolset/databases/mongodb/mongodb.py]: Why?

**Comment by nbrady-techempower** [comment]
@msmith-techempower databases still only start for tests that require them. I’m on mobile right now so not easy to link that to you but it’s unchanged

File: toolset/databases/mongodb/mongodb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nbrady-techempower) [reply toolset/databases/mongodb/mongodb.py]: Because that’s how you wrote it

**Comment by msmith-techempower** [comment]
> @msmith-techempower databases still only start for tests that require them. I’m on mobile right now so not easy to link that to you but it’s unchanged

Can you not simply take a vacation? Geeeez....

Okay, I read more code while not on a call, and I tend to agree with you.

File: toolset/databases/mongodb/mongodb.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (msmith-techempower) [reply toolset/databases/mongodb/mongodb.py]: I see no value in describing the database implementation, and if past me wrote that code then he probably had some reason, but he was also dumber than present me.

**Comment by nbrady-techempower** [comment]
Every day of my life is a vacation.

File: toolset/utils/metadata.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ class Metadata:
 
     supported_dbs = []
     for name in databases:
-        supported_dbs.append((name, databases[name].description))
+        supported_dbs.append((name, '...'))
```
Comment (msmith-techempower) [inline toolset/utils/metadata.py]: Is the second part of this tuple ever read? It should probably be removed and anything that relies on it made aware of that change.

Comment (nbrady-techempower) [reply toolset/utils/metadata.py]: Yes. The `allowed` property for all items in the benchmark config are expected to be tuples, not just databases. Makes it easier to check each benchmark config property with the same code: https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/toolset/utils/metadata.py#L372

A couple of other properties are built the same way: https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/toolset/utils/metadata.py#L295

**Pull request merged**